src/sage/matrix/matrix_rational_sparse.pyx

```diff
@@ -819,8 +828,74 @@ cdef class Matrix_rational_sparse(Matrix_sparse):
             [1 0 0]
             [0 1 0]
             [0 0 1]
+
+        .. SEEALSO::
+
+            :meth:`~sage.matrix.matrix_rational_dense.Matrix_rational_dense._right_kernel_matrix`
+        """
+        algorithm = kwds.get('algorithm', 'default')
+        if algorithm == 'default' or algorithm == 'iml':
+            return self.dense_matrix()._right_kernel_matrix()
+        elif algorithm == 'linbox':
+            return self._right_kernel_matrix_linbox()
+        else:
+            raise ValueError("matrix kernel algorithm '%s' not recognized" % algorithm)
+
+    def _right_kernel_matrix_linbox(self):
+        r"""
+        Return a pair that includes a matrix of basis vectors
+        for the right kernel of ``self``.
+
+        OUTPUT:
+
+        Returns a pair.  First item is the string 'computed-linbox-rational'
+        that identifies the nature of the basis vectors.
+
+        Second item is a matrix whose rows are a basis for the right kernel,
+        over the rationals, as computed by the LinBox library.
         """
-        return self.dense_matrix()._right_kernel_matrix()
+        # NOTE: Degenerate cases (0 rows or columns) are handled by right_kernel_matrix.
+
+        cdef givaro.QField givQQ
+        cdef linbox.SparseMatrix_rational * M = new_linbox_matrix_rational_sparse(givQQ, self)
+
+        MQ = sage.matrix.matrix_space.MatrixSpace(QQ, self._ncols, self._ncols, sparse=True)
+        A = MQ.zero_matrix().__copy__()
+
+        cdef linbox.SparseMatrix_rational * N = new_linbox_matrix_rational_sparse(givQQ, A)
+
+        cdef linbox.GaussDomain_rational * dom = new linbox.GaussDomain_rational(givQQ)
```

Comment:
I don't think you need to place `dom` or `N` or `M` in the heap, maybe it's fine to just place them on the stack?